asf_public_discourse_home_decarbonisation/analysis/buildhub/extract_faqs.py

```diff
@@ -0,0 +1,245 @@
+"""
+This script is designed to extract frequently asked questions (FAQs) and specific patterns from a given forum text dataset.
+Key Functionalities:
+    - Extract Questions: Identifies and extracts questions from text based on criteria such as the number of words and minimum sentence length.
+    - Extract Specific Patterns: Targets and extracts specific patterns or phrases, tailored by the `extract_idk` function.
+    - Configurable: Allows runtime configuration through command-line arguments for flexible usage in different scenarios.
+To run this script:
+    'python extract_faqs.py --forum <forum> --category <category> --collection_date_time <collection_date_time>'
+where:
+    <forum>: The forum that we are looking at (e.g. bh or MSE).
+    <category>: The category within the forum (e.g. energy).
+    <collection_date_time>: The date time format, this differs between 'YYYY_MM_DD' for MSE and 'YYMMDD' for bh.
+Example Usage:
+    For MSE:
+        'python extract_faqs.py --forum mse --category energy --collection_date_time 2023_11_15'
+    For buildhub:
+        'python extract_faqs.py --forum bh --category combined_data --collection_date_time 231120'
+    Note: In these run commands note the different format in date-time format for the different forums.
+"""
+import pandas as pd
+import re
+import nltk
+from nltk.tokenize import sent_tokenize
+from typing import List, Tuple
+import argparse
+from asf_public_discourse_home_decarbonisation.getters.bh_getters import (
+    get_bh_category_data,
+)
+from asf_public_discourse_home_decarbonisation.getters.mse_getters import (
+    get_mse_category_data,
```

Comment:
If you use this getter instead, it will allow to read category data as well as data from all categories at once.
```suggestion
from asf_public_discourse_home_decarbonisation.getters.mse_getters import (
    get_mse_data,
```

Reply:
I guess the way the script is currently functioning is that we are looking at a single category at a time (where all the data can be classified as a category of "all"). Would the functionality of this mean we would get the questions for a specific category as well as all the data? Potentially calculate the percentage of questions within that given category?

Reply:
Discussed during stand-up, now I've understood it and going to make this simple change.